vector/src/main/java/im/vector/RegistrationManager.java

```diff
@@ -199,23 +199,19 @@ public void setSupportedRegistrationFlows(final RegistrationFlowResponse registr
      */
     public void checkUsernameAvailability(final Context context, final UsernameValidityListener listener) {
         if (getLoginRestClient() != null) {
-            final RegistrationParams params = new RegistrationParams();
+            // Trigger a fake registration (without password) to know whether the user name is available or not.
+            RegistrationParams params = new RegistrationParams();
             params.username = mUsername;
-            params.password = mPassword;
 
             register(context, params, new InternalRegistrationListener() {
                 @Override
                 public void onRegistrationSuccess() {
-                    listener.onUsernameAvailabilityChecked(true);
+                    // The registration could not succeed without password.
```

Comment:
It's quite dangerous to remove the callback in this case (trying to avoid infinite loading view)
You should keep calling `listener.onUsernameAvailabilityChecked(false);` (the error case) as a fallback, for symetry